### PR TITLE
Add support for GetLength and SetLength

### PR DESCRIPTION
**Pull request by fivedots**
Also add a short example of usage to wrapped.cpp

### CODE REVIEW

File: nativeio_async_wrapper.js [hunk 1]
```diff
@@ -180,6 +192,45 @@ mergeInto(LibraryManager.library, {
     })
   },
 
+  NativeIO_SetLength__deps: ['$NativeIOWrapper','$NativeIOUtils', '$Asyncify'],
+  NativeIO_SetLength: function(fileDescriptor, {{{ defineI64Param('length') }}}) {
+    return Asyncify.handleSleep(function(wakeUp) {
+      {{{ receiveI64ParamAsI32s('length') }}}
+      try {
```
Comment (rstz) [inline nativeio_async_wrapper.js:199]: Could you add a comment why this is necessary?

File: nativeio_async_wrapper.js [hunk 2]
```diff
@@ -180,6 +192,45 @@ mergeInto(LibraryManager.library, {
...
+        var length = NativeIOUtils.combineI64Params(length_low, length_high);
```
Comment (rstz) [inline nativeio_async_wrapper.js:202]: Could you add a comment why this is necessary?

File: nativeio_async_wrapper.js [hunk 3]
```diff
@@ -180,6 +192,45 @@ mergeInto(LibraryManager.library, {
...
+           fileDescriptor + ':', error);
+        wakeUp(-{{{cDefine('EINVAL')}}});
+      }
+
+      NativeIOWrapper.setLength(fileDescriptor, length).then(
+        () => {wakeUp(0)},
+        (error) => {
+          console.log(
+            'NativeIO error while setting length of file with file descriptor number',
+            fileDescriptor, 'to', offset + ':', error);
```
Comment (rstz) [inline nativeio_async_wrapper.js]: Nit: you could use ` for string formatting here as well

File: nativeio_async_wrapper.js [hunk 4]
```diff
@@ -180,6 +192,45 @@ mergeInto(LibraryManager.library, {
...
+        })
+    })
+  },
+
+  NativeIO_GetLength__deps: ['$NativeIOWrapper','$NativeIOUtils', '$Asyncify'],
+  NativeIO_GetLength: function(fileDescriptor) {
+    return Asyncify.handleSleep(function(wakeUp) {
+      NativeIOWrapper.getLength(fileDescriptor).then(
+        (length) => {
+          wakeUp(length)},
```
Comment (rstz) [inline nativeio_async_wrapper.js]: Nit: line break after } ?

File: nativeio_async_wrapper.js [hunk 5]
```diff
@@ -76,7 +88,7 @@ mergeInto(LibraryManager.library, {
       NativeIOWrapper.open(name).then(
         (fd) => {wakeUp(fd)},
         (error) => {
-          console.log('NativeIO error while opening \"', name, '\": ', error);
+          console.log('NativeIO error while opening \"'+ name + '\": ', error);
```
Comment (rstz) [inline nativeio_async_wrapper.js]: I think a more readable variant is
```suggestion
          console.log(`NativeIO error while opening "${name}": ${error}`);
```

File: nativeio_async_wrapper.js (hunk 3 above)
Comment (fivedots) [reply nativeio_async_wrapper.js]: Done, as well as doing it for all lines were params were used within an error message.

File: nativeio_async_wrapper.js (hunk 5 above)
Comment (fivedots) [reply nativeio_async_wrapper.js]: Done.

File: nativeio_async_wrapper.js (hunk 4 above)
Comment (fivedots) [reply nativeio_async_wrapper.js]: I moved everything into one line for consistency.

File: nativeio_async_wrapper.js (hunk 2 above)
Comment (fivedots) [reply nativeio_async_wrapper.js:202]: Added a comment to function.

**Review by rstz — APPROVED** [review]

**Pull request merged**